troubleshooting/rocksdb/src/provided/java/com/ververica/flink/training/provided/RocksDBTuningJob1.java

```diff
@@ -0,0 +1,170 @@
+package com.ververica.flink.training.provided;
+
+import org.apache.flink.api.common.eventtime.WatermarkStrategy;
+import org.apache.flink.api.common.state.MapState;
+import org.apache.flink.api.common.state.MapStateDescriptor;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.utils.ParameterTool;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.TimerService;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.KeyedProcessFunction;
+import org.apache.flink.streaming.api.functions.sink.DiscardingSink;
+import org.apache.flink.util.Collector;
+
+import com.ververica.flink.training.common.DoNotChangeThis;
+import com.ververica.flink.training.common.Measurement;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+import static com.ververica.flink.training.common.EnvironmentUtils.createConfiguredEnvironment;
+import static com.ververica.flink.training.common.EnvironmentUtils.isLocal;
+
+/** RocksDB Training job. */
+@DoNotChangeThis
+public class RocksDBTuningJob1 {
+
+    /**
+     * Creates and starts the troubled RocksDB job.
+     *
+     * @throws Exception if the application is misconfigured or fails during job submission
+     */
+    public static void main(String[] args) throws Exception {
+        ParameterTool parameters = ParameterTool.fromArgs(args);
+
+        int numSensors = parameters.getInt("numSensors", 100_000);
+        int payloadWords = parameters.getInt("payloadWords", 1_000);
+        int lastNTimestamps = parameters.getInt("lastNTimestamps", 5);
+        if (lastNTimestamps < 1) {
+            throw new IllegalArgumentException(
+                    "Must at least use one event for computing the result");
+        }
+
+        StreamExecutionEnvironment env = createConfiguredEnvironment(parameters);
+
+        // Checkpointing Configuration (use cluster-configs if not run locally)
+        if (isLocal(parameters)) {
```

Comment:
Just commenting to say 1.15 will support disabling in code configs and I cannot wait

Reply:
I hope that it does at least print messages in the log like "ignoring code-config xxxx" for the user to know which of these were not applied...it's already hard enough to get the final overview of which option is set and which is not

Reply:
Agreed